plugin/src/main/java/org/opensearch/ml/action/connector/UpdateConnectorTransportAction.java

```diff
@@ -0,0 +1,146 @@
+        this.client = client;
+        this.connectorAccessControlHelper = connectorAccessControlHelper;
+        this.mlModelManager = mlModelManager;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<UpdateResponse> listener) {
+        MLUpdateConnectorRequest mlUpdateConnectorAction = MLUpdateConnectorRequest.fromActionRequest(request);
+        String connectorId = mlUpdateConnectorAction.getConnectorId();
+        UpdateRequest updateRequest = new UpdateRequest(ML_CONNECTOR_INDEX, connectorId);
+        updateRequest.doc(mlUpdateConnectorAction.getUpdateContent());
+        updateRequest.docAsUpsert(true);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            connectorAccessControlHelper.validateConnectorAccess(client, connectorId, ActionListener.wrap(hasPermission -> {
+                if (Boolean.TRUE.equals(hasPermission)) {
+                    updateUndeployedConnector(connectorId, updateRequest, listener, context);
+                } else {
+                    listener
+                        .onFailure(
+                            new IllegalArgumentException("You don't have permission to update the connector, connector id: " + connectorId)
+                        );
+                }
+            }, exception -> {
+                log.error("Permission denied: Unable to update the connector with ID {}. Details: {}", connectorId, exception);
+                listener.onFailure(exception);
+            }));
+        } catch (Exception e) {
+            log.error("Failed to update ML connector for connector id {}. Details {}:", connectorId, e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void updateUndeployedConnector(
+        String connectorId,
+        UpdateRequest updateRequest,
+        ActionListener<UpdateResponse> listener,
+        ThreadContext.StoredContext context
+    ) {
+        SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX);
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
+        boolQueryBuilder.must(QueryBuilders.matchQuery(MLModel.CONNECTOR_ID_FIELD, connectorId));
+        boolQueryBuilder.must(QueryBuilders.idsQuery().addIds(mlModelManager.getAllModelIds()));
+        sourceBuilder.query(boolQueryBuilder);
+        searchRequest.source(sourceBuilder);
+
+        client.search(searchRequest, ActionListener.wrap(searchResponse -> {
+            SearchHit[] searchHits = searchResponse.getHits().getHits();
+            if (searchHits.length == 0) {
+                client.update(updateRequest, getUpdateResponseListener(connectorId, listener, context));
+            } else {
+                log.error(searchHits.length + " models are still using this connector, please undeploy the models first!");
```

Comment:
I feel like best user experience will be, if we show the deployed model ids also to the customer. Otherwise with this message user again needs to query to find out the deployed models.

Customer might not have any idea which models are deployed now with this connector idea. If we show the model ids, that could be helpful for customer.

Reply:
line 100 will use all the deployed and deploying models to search ML-Model index that match "connector_id=<updated connector id>". The model id shouldn't be included in the message otherwise app sec reviewer will raise concerns about the customer info leakage in the logs.

Reply:
not sure how a random model id can concern appsec which is not quite a customer info. 
Anyway, I was mostly referring to the `new MLValidationException(` part where we can send these model ids.

The goal is to make this easy for customers. That shouldn't concern appsec at least.

Reply:
> `all the deployed and deploying models`

does this get a inclusive list of all the deployed and deploying models in `all the nodes`?